### PR TITLE
Generate REST Client Implementations

### PR DESCRIPTION
**Pull request by JianpingChen**
https://github.com/Azure/azure-sdk-for-android/issues/651

### CODE REVIEW

**Comment by anuchandy** [comment]
@srnagar @jianghaolu could you guys please take a look at the refactoring in autorest java to accommodate the android generation.

File: androidgen/src/main/java/com/azure/autorest/android/template/AndroidProxyTemplate.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        interfaceBlock.annotation(annotationBuilder.toString());
+    }
+/*
+    @Override
+    protected void writeProxyMethodSignature(ArrayList<String> parameterDeclarationList, ProxyMethod restAPIMethod, JavaInterface interfaceBlock) {
+        String parameterDeclarations = String.join(", ", parameterDeclarationList);
+        IType restAPIMethodReturnValueClientType = restAPIMethod.getReturnType().getClientType();
+        if(restAPIMethodReturnValueClientType instanceof GenericType) {
+            GenericType genericReturnType = (GenericType) restAPIMethodReturnValueClientType;
+            if (genericReturnType.getName().startsWith("Mono")) {
```
Comment (JianpingChen) [inline androidgen/src/main/java/com/azure/autorest/android/template/AndroidProxyTemplate.java:52]: Should not reference "Mono" here

**Comment by weidongxu-microsoft** [comment]
Maybe not my part. Just asking.

Is the base branch same as `v4`? Does CI passes?

File: androidgen/src/main/java/com/azure/autorest/android/template/AndroidProxyTemplate.java (hunk 1 above)
Comment (JianpingChen) [reply androidgen/src/main/java/com/azure/autorest/android/template/AndroidProxyTemplate.java:52]: Never mind. this part is commented out.

**Comment by JianpingChen** [comment]
> Maybe not my part. Just asking.
> 
> Is the base branch same as `v4`? Does CI passes?

Yes. autorest_v4_android_v2 is branched off v4. I don't think there is a CI pipeline set up for this.

**Pull request closed without merging**